File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    arr = []
    
    for i in range(len(progresses)):
        if (100 - progresses[i]) % speeds[i] == 0:
            arr.append(int((100 - progresses[i]) // speeds[i]))
        else:
            arr.append(int(((100 - progresses[i]) // speeds[i])+1))
    
    m = arr[0]
    cnt = 1
    for j in range(1,len(arr)):
        if arr[j] > m:
            m = arr[j]
            answer.append(cnt)
            cnt = 1
        else:
            cnt += 1
    answer.append(cnt)
    # return arr
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] remain = new int[speeds.length];
        for(int i=0; i<speeds.length; i++){
            int temp = 100 - progresses[i];
            if(temp % speeds[i] == 0){
                remain[i] = (temp / speeds[i]);
            } else {
                remain[i] = (temp / speeds[i])+1;
            }
        }
        List<Integer> answer = new ArrayList<>();
        int t = remain[0];
        int cnt = 1;
        for(int i=1; i<remain.length; i++){
            if(remain[i] > t){
                answer.add(cnt);
                t = remain[i];
                cnt = 1;  
            } else {
                cnt++;
            }
        }
        answer.add(cnt);
        // System.out.println(answer);
        int[] ans = new int[answer.size()];
        for(int i=0; i<answer.size(); i++){
            ans[i] = answer.get(i);
        }
        return ans;
    }
}